daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Pretty.scala

```diff
@@ -31,10 +31,23 @@ import com.daml.lf.speedy.SBuiltin._
 
 private[lf] object Pretty {
 
-  def prettyError(err: SError, ptx: PartialTransaction): Doc =
+  private[this] val dummyTx = PartialTransaction.initial(
+    (_ => TransactionVersion.minVersion),
+    ContractKeyUniquenessMode.On,
+    submissionTime = Time.Timestamp.MinValue,
+    initialSeeds = InitialSeeding.NoSeed,
+  )
+
+  def prettyError(err: SError, ptx: PartialTransaction = dummyTx): Doc =
+    // TODO https://github.com/digital-asset/daml/issues/9974
+    //  (MK) I’m not convinced the partial transaction here provides any real value.
+    //  Afaict the only thing we use it for is to produce slightly different error
+    //  if a contract is inactive due to a recursive exercise. We don’t do that in
+    //  the scenario service however and I’ve never heard a request for it so I
+    //  have a hard time believing that it is really useful here.
```

Comment:
Added this to https://github.com/digital-asset/daml/issues/9974 's description, feel free to delete [the comment](https://github.com/digital-asset/daml/issues/9974#issuecomment-867649444) if unnecessary. 🙂